src/test/java/org/javarosa/xpath/expr/EncodingDecodeTest.java

```diff
@@ -0,0 +1,41 @@
+package org.javarosa.xpath.expr;
+
+import static org.javarosa.xpath.expr.Encoding.BASE64;
+import static org.javarosa.xpath.expr.Encoding.HEX;
+import static org.junit.Assert.assertEquals;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class EncodingDecodeTest {
```

Comment:
I have a pretty strong preference to test at both levels because it's a pattern that exists throughout JR. There have been cases of functions that work when tested in isolation like this but not in forms. It's helpful to have the test coverage to differentiate the two, especially when there are regressions.

In this specific case, I further like the symmetry with `EncodingEncodeTest`.